src/core/postNote.ts

```diff
@@ -0,0 +1,146 @@
+import joplin from 'api';
+import {Attachment} from '../model/attachment.model';
+import {EmailContent} from '../model/emailContent.model';
+import {EXPORT_TYPE} from '../constants';
+import {Attachments} from './postAttachments';
+import {ExportNoteBody} from '../model/exportNoteBdoy.model';
+import {AttachmentProperties} from 'src/model/attachmentProperties.mode';
+
+export class PostNote {
+    emailHtmlBody: string;
+    subject: string;
+    attachments: Attachment[];
+    folders: string[] = [];
+    tags: string[] = [];
+    noteBody: ExportNoteBody = {title: '', parent_id: '', body: '', body_html: '', markup_language: 1};
+
+    async createFolder(folder: string): Promise<void> {
+        await joplin.data.post(['folders'], null, {title: folder});
+    }
+
+    async createTag(tag: string): Promise<void> {
+        await joplin.data.post(['tags'], null, {title: tag});
+    }
+
+    async addFolders(folders: string[]): Promise<void> {
+        this.folders = folders;
+        const joplinFolders = await joplin.data.get(['folders']);
+
+        // Search by folder title
+        const joplinFoldersTitles = joplinFolders.items.map((joplinFolder)=> joplinFolder.title);
```

Comment:
It is possible to have multiple folders with the same name because they can be nested, so this logic may not always work correctly.
However, I don't see an easy fix right now. Definitely something to keep in mind though.

Reply:
I fixed it by searching by id and showing the path of folders in options.

https://user-images.githubusercontent.com/58605547/185251175-d45affa9-369d-4982-ac89-14d36e6f3cff.mp4